errcheck's -ignoretests flag is being ignored?

### What did you do?

Executed `:GoErrCheck` with various attempts at passing a valid flag for that command:

```
:GoErrCheck! -ignoretests
:GoErrCheck! -ignoretests %
:GoErrCheck! '-ignoretests'
:GoErrCheck! '-ignoretests' % 
```

### What did you expect to happen?

I expected to see a quickfix list with a list of errors that did not include test files.

Example of the flag working perfectly when executed via my terminal shell...

```
$ errcheck
base.go:917:16: writer.Close()
proxy_test.go:168:25:   defer resp.Body.Close()
proxy_test.go:254:25:   defer resp.Body.Close()
proxy_test.go:319:23:   defer resp.Body.Close()
proxy_test.go:341:23:   defer resp.Body.Close()
proxy_test.go:363:23:   defer resp.Body.Close()
proxy_test.go:376:14:   fmt.Fprintf(w, "uh-oh!")
proxy_test.go:414:25:   defer resp.Body.Close()
                                                                                                                                                                                                $ errcheck -ignoretests
base.go:917:16: writer.Close()
```

### What happened instead?

I got a quickfix list with a list of errors that included test files.

### Configuration

```
Plug 'fatih/vim-go', { 'do': ':GoUpdateBinaries' }

...

" ALE linting
"
" go vet options depends on installing extra command:
"
"   $ go install golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow
"
" ...just be sure to run that install outside of your projects directory,
" otherwise it'll add dependencies to your project go.mod accidentally.
"
let g:ale_go_govet_options = '-vettool=$(which shadow)'
let g:ale_linters = {'go': ['gopls']}
let g:ale_python_mypy_options = '--ignore-missing-imports --strict-equality'
let g:ale_sign_error = '✗'
let g:ale_sign_warning = '▲'
highlight link ALEWarningSign String
highlight link ALEErrorSign Title

nmap <silent> <leader>x :ALENext<cr>
nmap <silent> <leader>z :ALEPrevious<cr>

" vim-go
let g:go_fmt_command = 'goimports'
let g:go_gopls_complete_unimported = 1
let g:go_gopls_staticcheck = 1
let g:go_metalinter_command='gopls'
let g:go_metalinter_deadline = '20s'

" vim-go: check if any expressions return an error type that aren't being handled
autocmd BufWritePost *.go :GoErrCheck! -ignoretests %
```

#### vim-go version:

v1.22

####  `vimrc` you used to reproduce (use a *minimal* vimrc with other plugins disabled; do not link to a 2,000 line vimrc):
<details><summary>vimrc</summary><br><pre>

set nocompatible
set nobackup
set nowritebackup
set noswapfile

call plug#begin('~/.vim/plugged')

Plug 'fatih/vim-go', { 'do': ':GoUpdateBinaries' }

call plug#end()

autocmd BufWritePost *.go :GoErrCheck! -ignoretests %

</pre></details>

#### Vim version (first three lines from `:version`):

```
VIM - Vi IMproved 8.1 (2018 May 18, compiled Aug 12 2019 11:40:34)
macOS version
Included patches: 1-1839
```

#### Go version (`go version`):

```
go version go1.13 darwin/amd64
```

#### Go environment
<details><summary><code>go env</code> Output:</summary><br><pre>
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/markmcdonnell/Library/Caches/go-build"
GOENV="/Users/markmcdonnell/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/markmcdonnell/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/Cellar/go/1.13/libexec"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.13/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/zk/_jgd9h392p55w9fd4lzt5dpw0000gp/T/go-build183558573=/tmp/go-build -gno-record-gcc-switches -fno-common"

</pre></details>

#### gopls version
<details><summary><code>gopls version</code> Output:</summary><br><pre>
<!-- gopls version -->
golang.org/x/tools/gopls v0.3.1
    golang.org/x/tools/gopls@v0.3.1 h1:yNTWrf4gc4Or0UecjOas5pzOa3BL0WDDyKDV4Wz5VaM=
</pre></details>

--- a/autoload/go/lint.vim
+++ b/autoload/go/lint.vim
@@ -223,18 +223,19 @@ endfunction
 function! go#lint#Errcheck(bang, ...) abort
   if a:0 == 0
     let l:import_path = go#package#ImportPath()
-    if import_path == -1
+    if l:import_path == -1
       call go#util#EchoError('package is not inside GOPATH src')
       return
     endif
+    let l:args = [l:import_path]
   else
-    let l:import_path = join(a:000, ' ')
+    let l:args = a:000
   endif
 
   call go#util#EchoProgress('[errcheck] analysing ...')
   redraw
 
-  let [l:out, l:err] = go#util#Exec([go#config#ErrcheckBin(), '-abspath', l:import_path])
+  let [l:out, l:err] = go#util#ExecInDir([go#config#ErrcheckBin(), '-abspath'] + l:args)
 
   let l:listtype = go#list#Type("GoErrCheck")
   if l:err != 0

--- a/autoload/go/lint_test.vim
+++ b/autoload/go/lint_test.vim
@@ -221,7 +221,7 @@ func! Test_Lint_GOPATH() abort
 
   let expected = [
           \ {'lnum': 5, 'bufnr': bufnr('%'), 'col': 1, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'exported function MissingDoc should have comment or be unexported'},
-          \ {'lnum': 5, 'bufnr': bufnr('%')+6, 'col': 1, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'exported function AlsoMissingDoc should have comment or be unexported'}
+          \ {'lnum': 5, 'bufnr': bufnr('%')+5, 'col': 1, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'exported function AlsoMissingDoc should have comment or be unexported'}
       \ ]
 
   let winnr = winnr()
@@ -249,7 +249,7 @@ func! Test_Lint_NullModule() abort
 
   let expected = [
           \ {'lnum': 5, 'bufnr': bufnr('%'), 'col': 1, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'exported function MissingDoc should have comment or be unexported'},
-          \ {'lnum': 5, 'bufnr': bufnr('%')+6, 'col': 1, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'exported function AlsoMissingDoc should have comment or be unexported'}
+          \ {'lnum': 5, 'bufnr': bufnr('%')+5, 'col': 1, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'exported function AlsoMissingDoc should have comment or be unexported'}
       \ ]
 
   let winnr = winnr()
@@ -267,6 +267,51 @@ func! Test_Lint_NullModule() abort
   endwhile
 
   call gotest#assert_quickfix(actual, expected)
+endfunc
+
+func! Test_Errcheck() abort
+  let RestoreGOPATH = go#util#SetEnv('GOPATH', fnamemodify(getcwd(), ':p') . 'test-fixtures/lint')
+  silent exe 'e ' . $GOPATH . '/src/errcheck/errcheck.go'
+
+  try
+    let l:bufnr = bufnr('')
+    let expected = [
+          \ {'lnum': 9, 'bufnr': bufnr(''), 'col': 9, 'pattern': '', 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'module': '', 'text': ":\tio.Copy(os.Stdout, os.Stdin)"},
+          \ {'lnum': 10, 'bufnr': bufnr('')+1, 'col': 9, 'pattern': '', 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'module': '', 'text': ":\tio.Copy(os.Stdout, os.Stdin)"},
+        \ ]
+
+    " clear the location lists
+    call setqflist([], 'r')
+
+    call go#lint#Errcheck(1)
+
+    call gotest#assert_quickfix(getqflist(), expected)
+    call assert_equal(l:bufnr, bufnr(''))
+  finally
+    call call(RestoreGOPATH, [])
+  endtry
+endfunc
+
+func! Test_Errcheck_options() abort
+  let RestoreGOPATH = go#util#SetEnv('GOPATH', fnamemodify(getcwd(), ':p') . 'test-fixtures/lint')
+  silent exe 'e ' . $GOPATH . '/src/errcheck/errcheck.go'
+
+  try
+    let l:bufnr = bufnr('')
+    let expected = [
+          \ {'lnum': 9, 'bufnr': bufnr(''), 'col': 9, 'pattern': '', 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'module': '', 'text': ":\tio.Copy(os.Stdout, os.Stdin)"},
+        \ ]
+
+    " clear the location lists
+    call setqflist([], 'r')
+
+    call go#lint#Errcheck(1, '-ignoretests')
+
+    call gotest#assert_quickfix(getqflist(), expected)
+    call assert_equal(l:bufnr, bufnr(''))
+  finally
+    call call(RestoreGOPATH, [])
+  endtry
 endfunc
 
 func! Test_Errcheck_compilererror() abort

--- a/autoload/go/test-fixtures/lint/src/errcheck/errcheck.go
+++ b/autoload/go/test-fixtures/lint/src/errcheck/errcheck.go
@@ -1,0 +1,10 @@
+package errcheck
+
+import (
+	"io"
+	"os"
+)
+
+func foo() {
+	io.Copy(os.Stdout, os.Stdin)
+}

--- a/autoload/go/test-fixtures/lint/src/errcheck/errcheck_test.go
+++ b/autoload/go/test-fixtures/lint/src/errcheck/errcheck_test.go
@@ -1,0 +1,11 @@
+package errcheck
+
+import (
+	"io"
+	"os"
+	"testing"
+)
+
+func TestFoo(t *testing.T) {
+	io.Copy(os.Stdout, os.Stdin)
+}

